src/pyk/kcfg/explore.py

```diff
@@ -313,3 +330,20 @@ def extend(
 
         else:
             raise ValueError('Unhandled case.')
+
+    def add_circularities_module(
+        self, old_module_name: str, new_module_name: str, circularities: Iterable[KClaim]
+    ) -> None:
+        max_priority: int = 1  # Maybe we should compute this dynamically?
+        kast_rules = [
+            KRule(body=c.body, requires=c.requires, ensures=c.ensures, att=KAtt({'priority': max_priority}))
+            for c in circularities
+        ]
```

Comment:
Could we do instead:

```suggestion
        kast_rules = [c.let(att=c.att.update({'priority': max_priority})) for c in circularities]
```

Reply:
We need also to change the type of the circularity from `KClaim` to `KRule`, but otherwise, we can preserve the attributes as you suggested.